pkg/volume/csi/csi_attacher.go

```diff
@@ -102,17 +102,11 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		glog.V(4).Info(log("attachment [%v] for volume [%v] created successfully", attachID, csiSource.VolumeHandle))
 	}
 
-	// probe for attachment update here
-	// NOTE: any error from waiting for attachment is logged only.  This is because
-	// the primary intent of the enclosing method is to create VolumeAttachment.
-	// DONOT return that error here as it is mitigated in attacher.WaitForAttach.
-	volAttachmentOK := true
 	if _, err := c.waitForVolumeAttachment(csiSource.VolumeHandle, attachID, csiTimeout); err != nil {
-		volAttachmentOK = false
-		glog.Error(log("attacher.Attach attempted to wait for attachment to be ready, but failed with: %v", err))
+		return "", err
```

Comment:
Could this cause API quota issues because now we could be retrying a lot of VolumeAttachment.Create() operations for objects that already exist?  Can we potentially check the informer cache first to see if we already created it?

Reply:
The `volAttachmentOK` var was mainly there for status indication.  This change should work as just like before.  @msau42 are you saying to check for pre-existence before attemting to create ?

Reply:
Yes, because this change can cause Attach() to retry more than before.

Reply:
That belongs to a separate PR

Reply:
So, I checked how to fix this polling with something smarter, but it turns out that CSI volume plugin has no access to VolumeAttachment shared informer and the fix won't be simple one-liner. I filled #64429 to keep track of this, however, I am not actively working on it.